core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,290 @@
+        List<Component> components = new ArrayList<>();
+        securitySchemeMap = openAPI.getComponents().getSecuritySchemes();
+        versionNumber = Long.parseLong(openAPI.getInfo().getVersion());
+        connectionName = openAPI.getInfo().getTitle();
+
+
+
+        Paths paths = openAPI.getPaths();
+
+        for (String pathName : paths.keySet()) {
+            PathItem path = paths.get(pathName);
+            Map<PathItem.HttpMethod, Operation> operations = path.readOperationsMap();
+
+            for (PathItem.HttpMethod httpMethod : operations.keySet()) {
+                components.addAll(initComponents(operations.get(httpMethod), httpMethod, pathName));
+            }
+
+        }
+
+       return components;
+    }
+
+    private  List<Component> initComponents(Operation operation, PathItem.HttpMethod operationName, String pathName) {
+        List<Component> components = new ArrayList<>();
+        if (operation != null) {
+
+            Set<String> securities = getSecurities(operation.getParameters(), operation.getSecurity());
+            Set<String> requestContents = getRequestContents(operation.getRequestBody());
+            Set<String> responseContents = getResponseContents(operation.getResponses());
+
+            List<HashMap<String, String>>  names = generateName(securities, requestContents, responseContents);
+
+            if (!names.isEmpty()) {
+                for (HashMap<String, String> partNames : names) {
+                    components.add(createComponent(partNames, operation, operationName, pathName));
+                }
+            } else {
+                components.add(createComponent(new HashMap(), operation, operationName, pathName));
+            }
+        }
+        return components;
+    }
+
+
+    private  Set<String> getSecurities(List<io.swagger.v3.oas.models.parameters.Parameter> parameters, List<SecurityRequirement> securitiesList) {
+        Set<String> securities = new HashSet<>();
+        if (parameters != null) {
+            securities = parameters.stream().map(parameter -> {
+                if (parameter.getIn() != null && securitySchemeMap.containsKey(parameter.getName())) {
+                    return parameter.getName();
+                }
+                return null;
+            }).filter(Objects::nonNull).collect(Collectors.toSet());
+        }
+        if (securitiesList != null) {
+            securities.addAll(securitiesList.stream().map(securityRequirement -> (String) securityRequirement.keySet().toArray()[0]).collect(Collectors.toSet()));
+        }
+        return securities;
+    }
+
+    private  Set<String> getRequestContents(RequestBody requestBody) {
+        if (requestBody != null && requestBody.getContent() != null) {
+            return requestBody.getContent().keySet();
+        }
+        return new HashSet<>();
+    }
+
+    private  Set<String> getResponseContents(ApiResponses apiResponses) {
+        if (apiResponses!= null) {
```

Comment:
Can it be null? wouldn't it be an empty map?

Reply:
With no responses key : 
attribute paths.'/pet'(put).responses is missing
API RESPONSE : null

Process finished with exit code 0

With empty responses values:

attribute paths.'/pet'(put).responses is not of type `object`
API RESPONSE : null

Process finished with exit code 0

It returns only null value